### ISSUE TITLE
Incorrect "unnecessary trailing comma comma" for annotation array parameter

### ISSUE THREAD

**Issue by lwasyl** [pinterest/ktlint#1379]
## Expected Behavior

The following file should not trigger errors:

```kotlin
import kotlin.reflect.KClass

@Foo(
    values = [
        Foo::class,
        Foo::class,
    ],
)
annotation class Foo(val values: Array<KClass<*>>)
```

## Observed Behavior
```
Foo.kt:7:6: Lint error > [experimental:trailing-comma] Unnecessary trailing comma before ")"
```

When using KtLint to format the file, it removes the comma after `]` producing 
```kotlin
import kotlin.reflect.KClass

@Foo(
    values = [
        Foo::class,
        Foo::class,
    ]
)
annotation class Foo(val values: Array<KClass<*>>)
```

which is IMO wrong, as the trailing comma that was removed is useful + it's added by IntelliJ when formatting the file regardless

## Your Environment

* Version of ktlint used: 0.44.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): `lintKotlin`
* Version of Gradle used (if applicable): 7.4
* Operating System and version: -
* Link to your project (if it's a public repository): -

**Comment by paul-dingemans** [comment]
@lwasyl This depends on your configuration. If you are using a `.editorconfig` file (recommended approach) then please check whether properties below have been set:
```
ij_kotlin_allow_trailing_comma_on_call_site=true
ij_kotlin_allow_trailing_comma=true
```

**Comment by lwasyl** [comment]
I have both of those options in my `.editorconfig` file, and IJ seems to respect it.
```properties
[*.{kt, kts, gradle}]
indent_size = 4
insert_final_newline = true
max_line_length = 140
ij_kotlin_allow_trailing_comma = true
ij_kotlin_allow_trailing_comma_on_call_site = true
```
